.github/workflows/reusable.yml

```
@@ -0,0 +1,37 @@
+          bun-version: ${{ inputs.bun-version }}
+
+      - name: Install dependencies
+        run: bun install
+
+      - name: Run all tests
+        id: smoke-test-1
+        run: bun test
+
+      - name: Set result output
+        id: set-result
+        run: echo "test_result=${{ steps.smoke-test-1.outcome }}" >> $GITHUB_OUTPUT
```

actionlint

shellcheck reported issue in this script: SC2086:info:1:57: Double quote to prevent globbing and word splitting

(shellcheck)